CorsixTH/Lua/entities/vip.lua

```diff
@@ -255,6 +255,22 @@ function Vip:evaluateEmergency(success)
   end
 end
 
+--! Evaluate a value based on an evaluation array.
+--!param value (number) Value to evaluate.
+--!param eval_table (table of {upper, value}) Evaluation array with increasing
+--! upper bound (leave out for the final entry), and the value to return if
+--! the provided value is at or below the upper bound.
+--!return Evaluation value.
+local function evaluateProperty(value, eval_table)
+  for _, boundary in ipairs(eval_table) do
+    if not boundary.upper or boundary.upper >= value then
+      return boundary.value
+    end
+  end
+  -- Should never get here, but just in case
+  return 0
```

Comment:
Should assert or print paranoia debug output here, don't want exception cases to be silent.